src/soundbase.cpp

```diff
@@ -247,10 +258,51 @@ void CSoundBase::ParseCommandLineArgument ( const QString& strMIDISetup )
             iCtrlMIDIChannel = slMIDIParams[0].toUInt();
         }
 
+        bool bSimple = true;
+
         // [offset for level]
         if ( slMIDIParams.count() >= 2 )
         {
-            iMIDIOffsetFader = slMIDIParams[1].toUInt();
+            int i = slMIDIParams[1].toUInt ( &bSimple );
+            if ( bSimple )
+                iMIDIOffsetFader = i;
+        }
+
+        if ( bSimple )
+        {
+            for ( int i = 0; i < MAX_NUM_CHANNELS; i++ )
+            {
+                if ( i + iMIDIOffsetFader > 127 )
+                    break;
+                aMidiCtls[i + iMIDIOffsetFader] = { EMidiCtlType::Fader, i };
+            }
+            return;
+        }
+
+        // We have named controllers
+
+        for ( int i = 1; i < slMIDIParams.count(); i++ )
+        {
+            QString sParm = slMIDIParams[i].trimmed();
+            if ( sParm.isEmpty() )
+                continue;
+
+            int iCtrl = QString ( sMidiCtlChar ).indexOf ( sParm[0] );
+            if ( iCtrl < 0 )
+                continue;
+            EMidiCtlType eTyp = static_cast<EMidiCtlType> ( iCtrl );
+
+            const QStringList slP = sParm.mid ( 1 ).split ( '*' );
+            int iFirst = slP[0].toUInt();
+            int iNum = ( slP.count() > 1 ) ? slP[1].toUInt() : 1;
+            for ( int iOff = 0; iOff < iNum; iOff++ )
+            {
+                if ( iOff >= MAX_NUM_CHANNELS )
+                    break;
```

Comment:
Similarly, this is syntax validation and, if it fails, should tell the user what went wrong.

Reply:
Same.